andela/zinnia-ah-backend
# 164797145 Build validation to protect user creation endpoint

#### What does this PR do?
Builds validation to protect user creation endpoint

#### Description of Task to be completed?
- [x] create mockdata to simulate invalid user inputs
- [x] create joi schema for user signup endpoint
- [x] create logic to handle validation errors

#### How should this be manually tested?
* clone repo at https://github.com/andela/zinnia-ah-backend.git
* run ```git checkout ft-handle-user-creation-validation-164797145```
* run `npm install`
* open a terminal inside project folder and run ``npm test``
* run npm test
* All tests should pass
* inside your terminal, run ```npm run start:dev```

#### Any background context you want to provide?
- Never trust the user to provide the right data inside your forms. Hence, validation gives descriptive feedback to guide the user

#### What are the relevant pivotal tracker stories? (if applicable)
164797145

#### Screenshots
None
